### ISSUE TITLE
Function composition using `<|` as  prefix operator 

### ISSUE THREAD

**Issue by brihat** [JuliaLang/julia#4963]
I saw that `|>` acts like a postfix operator. That is, `x |> f` is translated to `f(x)`. In the same vein, it would be intuitive to define `<|` as a prefix operator. That is `f <| x` translates to `f(x)`. This enables some nice function chaining:

```
julia> log <| exp <| 10
         10.0
```

This is similar to the `@` operator in [Mathematica](http://reference.wolfram.com/mathematica/ref/Prefix.html). It's quite easy to define this behaviour:

```
(<|)(f::Function, x) = f(x)
(<|)(f::Function,g::Function) = x->f(g(x))
```

I was wondering what other people think about this feature. Myself and  @gitfoxi discovered this possibility in IRC. If it's useful, one of us can submit a pull request.

**Comment by hayd** [comment]
Shouldn't this be circ rather than `<|`:

    g∘f(x) == g(f(x))

i.e.
```
(sin ∘ log ∘ f)(x^2 + x)
# rather than
sin <| log <| f |< (x^2 + x)
```
IMO this is not worth doing if it's not going to be as efficient as the parenthesis long-hand e.g. not defined using lambdas. That was partially why compose was previously removed: https://github.com/JuliaLang/julia/issues/2485#issuecomment-14525949.

**Comment by milktrader** [comment]
Love circ, how is that entered in REPL (pretty sure this is platform dependent)

**Comment by kmsquire** [comment]
`\circ<tab>`

**Comment by milktrader** [comment]
ah, nice!

**Comment by oschulz** [comment]
`\circ` seems perfect for function composition, I'd love to have that in Base.

**Comment by okvs** [comment]
This seems to work already:
```
> ∘(f::Function, g::Function) = x->f(g(x))
> r = randn(10)`
> map(sqrt∘abs∘sin, r) == map(x->sqrt(abs(sin(x))), r)
true
```

**Comment by hayd** [comment]
Now that we have #13412 this seems a reasonable thing to define in Base and export.

**Comment by fredrikekre** [comment]
Solved by #17155

**Comment by DilumAluthge** [comment]
>Solved by #17155

Can this issue be closed?
